### PR TITLE
Validate full path of file instead of file name in parsing /resource args

### LINKED ISSUE #27045: Wrong CS2021 is reported for *.resources files
**Investigation**
A change in src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs made in https://github.com/dotnet/roslyn/commit/d23cc53f65501798ae46e8ab9f3aa74077a7586d causes a bug.

There, PathUtilities.IsValidFilePath is called for a fileName instead of a filePath. Inside it,
new FileInfo(fileName) is created, and there is no parent directory in fileName, so current directory is used. In our case current directory is a path to compiler binaries, and it is rather long. If path to compiler binaries length + file name length exceed 260 characters, then PathTooLongException is thrown and that results in CS2021 error.

Compiler errors should not depend on a path to compiler binaries, only on path to files being compiled.

**Version Used**: 
2.8 (from dev15.7)

**Steps to Reproduce**:
1. Place compiler binaries into a folder with a path length >=127 characters (say, C:\Users\SomeUserAtSomeCompany\AppData\Local\MyFolder1\~\Toolset\A367E15E0EE028AF\MyFolder1.Toolset.Roslyn.2.8.20180511.192024)
2. Make a resource file with a name length >= 135 characters (say, MyCompany.SomeNamespace.SubNamespace.MoreNamespace.OneMoreNamespace.EvenMoreNamespace.Control.ShowMeSomeOptionsOptionsPagePanel.resources) and place it in your project with a short folder name so that a full path length to resource file would not exceed 260 characters.
3. Compile your project with a said resource file using a compiler run from its folder from step 1.

**Expected Behavior**:
No errors are produced

**Actual Behavior**:
CSC : error CS2021: File name '...\MyCompany.SomeNamespace.SubNamespace.MoreNamespace.OneMoreNamespace.EvenMoreNamespace.Control.ShowMeSomeOptionsOptionsPagePanel.resources' is empty, contains invalid characters, has a drive specification without an absolute path, or is too long

### PR DESCRIPTION
**Pull request by OmarTawfik**
Fixes #27045 

The original bug is that the file name was checked instead of the full path. This makes the `FileInfo` constructor consider the execution path as the base path for the (relative) file name.

This change has no tests associated, as I talked with a couple of people on the team, and I believe this case is exteremely hard to put in a unit test, as the test needs to move the executed tests/compiler somewhere long enough to repro, and the scenario is corner case enough I don't think it is worth modifying the test harness at this point.

### CODE REVIEW

**Comment by jaredpar** [comment]
approved

File: src/Compilers/Core/Portable/FileSystem/PathUtilities.cs [hunk 1]
```diff
@@ -697,18 +697,16 @@ public static string NormalizePathPrefix(string filePath, ImmutableArray<KeyValu
         /// 
         /// The more accurate way is to let the framework parse the path and throw on any errors.
         /// </summary>
-        /// <param name="path"></param>
-        /// <returns></returns>
-        public static bool IsValidFilePath(string path)
+        public static bool IsValidFilePath(string fullPath)
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/FileSystem/PathUtilities.cs]: >public static bool IsValidFilePath(string fullPath) [](start = 8, length = 51)

Would it be valid to add an assert ```Debug.Assert(IsAbsolute(fullPath));```?

File: src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs [hunk 2]
```diff
@@ -1796,7 +1796,7 @@ private static IEnumerable<InstrumentationKind> ParseInstrumentationKinds(string
                 return null;
             }
 
-            if (fullPath == null || !PathUtilities.IsValidFilePath(fileName))
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs]: >fullPath == null [](start = 16, length = 16)

No need to check for null?

File: src/Compilers/VisualBasic/Portable/CommandLine/VisualBasicCommandLineParser.vb [hunk 3]
```diff
@@ -1625,7 +1625,7 @@ lVbRuntimePlus:
                 Return Nothing
             End If
 
-            If fullPath Is Nothing OrElse Not PathUtilities.IsValidFilePath(fileName) Then
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/CommandLine/VisualBasicCommandLineParser.vb]: >fullPath Is Nothing OrElse  [](start = 15, length = 27)

No need to check for null?

**Comment by AlekseyTs** [comment]
Done with review pass (iteration 1)

File: src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs (hunk 2 above)
Comment (OmarTawfik) [reply src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs]: `IsValidFilePath` does that check.

---
In reply to: [192184799](https://github.com/dotnet/roslyn/pull/27279#discussion_r192184799) [](ancestors = 192184799)

File: src/Compilers/VisualBasic/Portable/CommandLine/VisualBasicCommandLineParser.vb (hunk 3 above)
Comment (OmarTawfik) [reply src/Compilers/VisualBasic/Portable/CommandLine/VisualBasicCommandLineParser.vb]: `IsValidFilePath` does that check.

---
In reply to: [192185024](https://github.com/dotnet/roslyn/pull/27279#discussion_r192185024) [](ancestors = 192185024)

File: src/Compilers/Core/Portable/FileSystem/PathUtilities.cs (hunk 1 above)
Comment (OmarTawfik) [reply src/Compilers/Core/Portable/FileSystem/PathUtilities.cs]: Yes. Will add that.

---
In reply to: [192184509](https://github.com/dotnet/roslyn/pull/27279#discussion_r192184509) [](ancestors = 192184509)

**Comment by OmarTawfik** [comment]
@dotnet-bot test this please

**Comment by OmarTawfik** [comment]
@AlekseyTs commented out your suggestion because of an existing bug.
Added a note to bring back the original `Debug.Assert()` when it is resolved.

**Pull request merged**